tests/Feature/Actions/Stripe/CancelSubscriptionTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Tests\Feature\Actions\Stripe;
+
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Laravel\Cashier\Subscription;
+use Mockery;
+use App\Actions\Stripe\CancelSubscription;
+use App\Models\User;
+use Tests\TestCase;
+
+class CancelSubscriptionTest extends TestCase
+{
+    use RefreshDatabase, WithFaker;
+    private CancelSubscription $action;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->action = new CancelSubscription();
+    }
+
+    protected function tearDown(): void
+    {
+        Mockery::close();
+        parent::tearDown();
+    }
+
+    public function test_it_returns_false_if_no_subscription_exists()
+    {
+        $user = Mockery::mock(User::class);
```

Comment:
@afaqahmad0051 We don't mock like this. please use functions provided by laravel rather than using underlying package directly.